src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstancesController.cs

```diff
@@ -581,6 +583,43 @@ public async Task<ActionResult<Instance>> Delete(int instanceOwnerPartyId, Guid
             return Ok(updatedInstance);
         }
 
+        /// <summary>
+        /// Updates the presentation fields of an instance
+        /// </summary>
+        [Authorize(Policy = AuthzConstants.POLICY_INSTANCE_WRITE)]
+        [HttpPut("{instanceOwnerPartyId:int}/{instanceGuid:guid}/presentationFields")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [HttpPut]
+        public async Task<Instance> UpdatePresentationFields(
+            [FromRoute] int instanceOwnerPartyId,
+            [FromRoute] Guid instanceGuid,
+            [FromBody] Dictionary<string, string> presentationFields)
+        {
+            string instanceId = $"{instanceOwnerPartyId}/{instanceGuid}";
+            Instance instance = await _instanceRepository.GetOne(instanceId, instanceOwnerPartyId);
+            Dictionary<string, string> existing = instance.PresentationFields;
+
+            if (existing == null)
+            {
+                instance.PresentationFields = new Dictionary<string, string>();
+            }
+
+            foreach (KeyValuePair<string, string> entry in presentationFields)
+            {
+                if (string.IsNullOrEmpty(entry.Value))
+                {
+                    instance.PresentationFields.Remove(entry.Key);
+                }
+                else
+                {
+                    instance.PresentationFields[entry.Key] = entry.Value;
```

Comment:
I don't think so.. it would have been identified by the unit test. We have test cases that should cover adding new keys and overwriting keys in an existing dictionary